cmd/drone-server/drone.go

```diff
@@ -236,6 +244,14 @@ func main() {
 		auth.POST("", server.GetLogin)
 	}
 
+	redirects := r.Group("/redirect")
+	{
+		redirects.Use(server.SetDatastore(store))
+		redirects.Use(server.SetRepo())
+
+		redirects.GET("/:owner/:name/commit/:sha", server.RedirectSha)
```

Comment:
great, thanks for the explanation!  Is it possible for GitLab to query the API prior to redirecting? GitLab could hit the Drone API, fetch the build by commit to get the build number, and then route the user to the appropriate URL?